pkg/controller/controller.go

```diff
@@ -42,10 +42,45 @@ import (
 	"k8s.io/utils/mount"
 )
 
+// RunLocalController facilitates and manages the sync loop.
+// It launches the main sync loop and if there is an updated configuration from the ConfigWatcher,
+// it will inform the main sync loop to terminate and then will launch a new sync loop with the
+// updated configuration.
+func RunLocalController(resync time.Duration, restartController <-chan common.ProvisionerConfiguration, client *kubernetes.Clientset, ptable deleter.ProcTable, discoveryPeriod time.Duration, node *v1.Node, namespace, jobImage string, config common.ProvisionerConfiguration) {
+	stopped := make(chan struct{})
+	stopChan := make(chan struct{})
+
+	defer func() {
+		close(stopped)
+		close(stopChan)
+	}()
+
+	go StartLocalController(stopChan, stopped, client, ptable, discoveryPeriod, common.UserConfigFromProvisionerConfig(node, namespace, jobImage, config))
+
+	for {
+		select {
+		case newConfig := <-restartController:
+			stopChan <- struct{}{}
+			<-stopped
+			go StartLocalController(stopChan, stopped, client, ptable, discoveryPeriod, common.UserConfigFromProvisionerConfig(node, namespace, jobImage, newConfig))
+		default:
```

Comment:
I think this method should block until we get a new config, do we need the default case?

Reply:
Yes you are right, no need the default case, fixed! Thanks